src/tree-view/utils.js

```diff
@@ -189,7 +189,71 @@ export const toggleIsExpanded = (
     return newNode;
   });
 };
-
+export const getCharMatchId = (
+  nodes: TreeNodeT<>[],
+  nodeId: TreeNodeIdT,
+  chars: string,
+  closestOmmer: TreeNodeIdT | null,
+  getId: (TreeNodeT<>) => TreeNodeIdT,
+) => {
+  var foundid = matchString(
+    nodes,
+    nodeId,
+    chars,
+    closestOmmer,
+    getId,
+    false,
+    true,
+  )[0];
+  if (!foundid)
+    foundid = matchString(
+      nodes,
+      nodeId,
+      chars,
+      closestOmmer,
+      getId,
+      false,
+      false,
+    )[0];
+  return foundid;
+};
+export const matchString = (
```

Comment:
It would be great to write some unit tests for function

Reply:
I will change that function. I combined range search into DFS . As suggested by Chase (https://github.com/uber/baseweb/pull/3560#discussion_r463752404), I will change that to a direct search.

Reply:
The e2e test looks good. Can you also add a unit test for this function like here? https://github.com/uber/baseweb/blob/b16b9845f984ed62413e0069aef719ea4c1ab1b3/src/tree-view/__tests__/utils.test.js